File: src/main/java/tech/gdragon/commands/settings/AliasCommand.java
package tech.gdragon.commands.settings;

import net.dv8tion.jda.core.events.message.guild.GuildMessageReceivedEvent;
import tech.gdragon.DiscordBot;
import tech.gdragon.commands.Command;
import tech.gdragon.commands.CommandHandler;

@Deprecated
public class AliasCommand implements Command {

  @Override
  public void action(String[] args, GuildMessageReceivedEvent e) {
    if (args.length != 2) {
      String prefix = DiscordBot.serverSettings.get(e.getGuild().getId()).prefix;
      DiscordBot.sendMessage(e.getChannel(), usage(prefix));
      return;
    }

    if (!CommandHandler.commands.containsKey(args[0].toLowerCase())) {
      DiscordBot.sendMessage(e.getChannel(), "Command '" + args[0].toLowerCase() + "' not found.");
      return;
    }

    if (CommandHandler.commands.containsValue(args[1].toLowerCase())) {
      DiscordBot.sendMessage(e.getChannel(), "Alias '" + args[1].toLowerCase() + "' already exists.");
      return;
    }

    DiscordBot.serverSettings.get(e.getGuild().getId()).aliases.put(args[1].toLowerCase(), args[0].toLowerCase());
    DiscordBot.writeSettingsJson();
    DiscordBot.sendMessage(e.getChannel(), "New alias '" + args[1].toLowerCase() + "' set for the command '" + args[0].toLowerCase() + "'.");

  }

  @Override
  public String usage(String prefix) {
    return prefix + "alias [command name] [new command alias]";
  }

  @Override
  public String description() {
    return "Creates an alias, or alternate name, to a command for customization.";
  }
}

File: src/main/kotlin/tech/gdragon/commands/settings/AliasCommand.kt
package tech.gdragon.commands.settings

import net.dv8tion.jda.core.events.message.guild.GuildMessageReceivedEvent
import org.jetbrains.exposed.sql.transactions.transaction
import tech.gdragon.BotUtils
import tech.gdragon.commands.Command
import tech.gdragon.commands.CommandHandler
import tech.gdragon.db.dao.Alias
import tech.gdragon.db.dao.Guild

class AliasCommand : Command {
  override fun action(args: Array<String>, event: GuildMessageReceivedEvent) {
    // Argument count must be two
    require(args.size == 2) {
      val prefix = transaction { Guild.findById(event.guild.idLong)!!.settings.prefix }
      BotUtils.sendMessage(event.channel, usage(prefix))
    }

    val channel = event.channel
    val command = args[0].toLowerCase()

    // Checks that command to alias exists
    if (!CommandHandler.commands.containsKey(command)) {
      BotUtils.sendMessage(channel, "Command '$command' not found.")
    } else {
      val aliases = transaction { Guild.findById(event.guild.idLong)!!.settings.aliases.toList() }
      val alias = args[1].toLowerCase()

      // Checks that alias doesn't already exist
      if (aliases.find { it.name == alias } != null) {
        BotUtils.sendMessage (channel, "Alias '$alias' already exists.")
      } else {
        transaction {
          Guild.findById(event.guild.idLong)?.settings?.let {
            Alias.new {
              name = command
              this.alias = alias
              settings = it
            }

            BotUtils.sendMessage(channel, "New alias '$alias' set for command '$command'.")
          }
        }
      }
    }
  }

  override fun usage(prefix: String): String = "${prefix}alias [command name] [new command alias]"

  override fun description(): String = "Creates an alias, or alternate name, to a command for customization."
}
